### PR TITLE
refactor: redesign KMS service

### LINKED ISSUE #144: Redesign storage and secret lock resolvers for Key Manager
Logic is too complicated, hard to maintain, and with poor test coverage - needs to be redesigned.

### PR DESCRIPTION
**Pull request by aholovko**
This PR is mainly about improving maintainability of the code.

It simplifies `startcmd` package by getting rid of storage and secret lock resolvers (their simplified variants went to the redesigned KMS service - the new `ResolveKeystore` method).

The KMS service was simplified as well and its API should be cleaner now (just crypto operations + functionality for resolving Keystore from HTTP request for the particular user).

The idea of Keystore was also updated. In the previous version it was represented by the `Keystore` struct and supporting service around it (`keystore.Service`). Now it's "something" that can `CreateKey`, `ExportKey` and `GetKeyHandle` by key ID. It's backed by the key manager that, in turn, can use EDV for storing keys and secret lock with split secret shares.

From a performance standpoint - there are some improvements (especially in create keystore handler) but they are not so noticeable as 90% of all time is taken by ZCAP and that part hasn't improved yet.

It's more like a first step toward making KMS more production-ready and is not the final design. There is still plenty of stuff that can/should be improved.

And sorry guys for so big PR - it affects key services and I don't know the quick way of doing that big refactoring in smaller chunks.

Closes #144

Signed-off-by: Andriy Holovko <andriy.holovko@gmail.com>

### CODE REVIEW

File: pkg/restapi/kms/operation/operations_cryptobox.go [hunk 1]
```diff
@@ -91,19 +100,21 @@ func (o *Operation) easyHandler(rw http.ResponseWriter, req *http.Request) { //n
 //        200: easyOpenResp
 //    default: errorResp
 func (o *Operation) easyOpenHandler(rw http.ResponseWriter, req *http.Request) { //nolint:funlen // TODO refactor
-	ctx, span := tracer.Start(req.Context(), "easyOpenHandler")
+	ctx, span := o.tracer.Start(req.Context(), "easyOpenHandler")
```
Comment (Baha-sk) [inline pkg/restapi/kms/operation/operations_cryptobox.go:103]: is the tracer logging any sensitive data? eg keys, decrypted ciphertext, etc.

Comment (aholovko) [reply pkg/restapi/kms/operation/operations_cryptobox.go:103]: nope, it's more about timing

**Review by Baha-sk — APPROVED** [review]
LGTM

File: pkg/restapi/kms/operation/operations_cryptobox.go (hunk 1 above)
Comment (aholovko) [reply pkg/restapi/kms/operation/operations_cryptobox.go:103]: also I plan to update that approach in the future - probably move it to the middleware
I don't like that instrumentation code messes with production code - it becomes really hard to read the business logic

**Review by troyronda — APPROVED** [review]

**Pull request merged**